draft-ietf-quic-http.md

```diff
@@ -450,15 +457,103 @@ field names MUST be converted to lowercase prior to their encoding.  A request
-{{malformed}}.
+the status code for the response.
+
+Pseudo-header fields are not HTTP header fields.  Endpoints MUST NOT generate
+pseudo-header fields other than those defined in this document, except as
+negotiated via an extension; see {{extensions}}.
+
+Pseudo-header fields are only valid in the context in which they are defined.
+Pseudo-header fields defined for requests MUST NOT appear in responses;
+pseudo-header fields defined for responses MUST NOT appear in requests.
+Pseudo-header fields MUST NOT appear in trailers.  Endpoints MUST treat a
+request or response that contains undefined or invalid pseudo-header fields as
+malformed ({{malformed}}).
+
+All pseudo-header fields MUST appear in the header block before regular header
+fields.  Any request or response that contains a pseudo-header field that
+appears in a header block after a regular header field MUST be treated as
+malformed ({{malformed}}).
+
+The following pseudo-header fields are defined for requests:
+
+  ":method":
+  : Contains the HTTP method ({{!RFC7231}}, Section 4)
+
+  ":scheme":
+  : Contains the scheme portion of the target URI ({{!RFC3986}}, Section 3.1)
+
+  : ":scheme" is not restricted to "http" and "https" schemed URIs.  A
+    proxy or gateway can translate requests for non-HTTP schemes,
+    enabling the use of HTTP to interact with non-HTTP services.
+
+  ":authority":
+
+  : Contains the authority portion of the target URI ([RFC3986], Section 3.2).
+    The authority MUST NOT include the deprecated "userinfo" subcomponent for
+    "http" or "https" schemed URIs.
+
+  : To ensure that the HTTP/1.1 request line can be reproduced
+    accurately, this pseudo-header field MUST be omitted when
+    translating from an HTTP/1.1 request that has a request target in
+    origin or asterisk form (see [RFC7230], Section 5.3).  Clients
+    that generate HTTP/3 requests directly SHOULD use the ":authority"
+    pseudo-header field instead of the Host header field.  An
+    intermediary that converts an HTTP/3 request to HTTP/1.1 MUST
+    create a Host header field if one is not present in a request by
+    copying the value of the ":authority" pseudo-header field.
+
+  ":path":
+
+  : Contains the path and query parts of the target URI (the "path-absolute"
+    production and optionally a '?' character followed by the "query"
+    production (see Sections 3.3 and 3.4 of [RFC3986]).  A request in asterisk
+    form includes the value '*' for the ":path" pseudo-header field.
+
+  : This pseudo-header field MUST NOT be empty for "http" or "https"
+    URIs; "http" or "https" URIs that do not contain a path component
+    MUST include a value of '/'.  The exception to this rule is an
+    OPTIONS request for an "http" or "https" URI that does not include
+    a path component; these MUST include a ":path" pseudo-header field
+    with a value of '*' (see [RFC7230], Section 5.3.4).
+
+All HTTP/3 requests MUST include exactly one valid value for the ":method",
```

Comment:
Arguably, this language from 7540 permits it to also contain any number of invalid values.  Worth rephrasing.